pkg/webhook/webhook.go

```diff
@@ -166,7 +167,30 @@ func (m *podMutator) Handle(ctx context.Context, req admission.Request) (respons
 		logger.Error("failed to marshal pod object", err)
 		return admission.Errored(http.StatusInternalServerError, err)
 	}
-	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledPod)
+
+	// patchResponse := admission.PatchResponseFromRaw(req.Object.Raw, marshaledPod)
+	// patches := []jsonpatch.JsonPatchOperation{}
+	// for _, patch := range patchResponse.Patches {
+	// 	// drop patches that are removing fields
+	// 	// this prevents patches that remove fields that the webhook doesn't know about
+	// 	// ref: https://github.com/Azure/azure-workload-identity/issues/1312
+	// 	if patch.Operation == "remove" {
```

Comment:
This commented out approach seems safer?  And maybe invert the check so we only keep the type of operations we understand?